src/frequenz/client/microgrid/_component_states.py

```diff
@@ -94,6 +94,6 @@ def from_pb(cls, evc_state: ev_charger.ComponentState) -> Self:
             Enum value corresponding to the protobuf message.
         """
         if not any(t.value == evc_state for t in EVChargerComponentState):
-            return cls(cls.UNSPECIFIED)
+            return cls(cls.UNKNOWN)
```

Comment:
I guess we can't escape having to repurpose something.  Now we are giving extra meaning to `UNKNOWN` instead, saying that either the modbus drivers couldn't recognize the state (and it gets set by the api service), or the client couldn't recognize the state.

Reply:
We can have `UNKNOWN_TO_CLIENT = -1` and because protobuf enums can't have negative numbers, it will always be unique, but might be overkill,  and hard to name.

Reply:
There are some enums that don't have a `UNKNOWN`, and for those, we're still returning UNSPECIFIED when a code is unknown to the client.  So maybe it is worth making them all consistent.

Reply:
Yeah, not sure. If you look at the gRPC errors, some codes can be produced by the server or client, like `INTERNAL` or `RESOURCE_EXHAUSTED` (see https://grpc.github.io/grpc/core/md_doc_statuscodes.html), I think we can see this similarly and say that the estate is unknown, I'm not sure if it is that relevant to the **user of the client** to know if it is unknown to the server or client, I can only see the case where the client is outdated compared to the API specs and the users somehow re-generate its own bindings and use that instead to see if the code is actually known by the server.

IMHO `UNKNOWN` is definitely more accurate here, and if some estate don't have this in the spec, I think that should be a specs bug. @tiyash-basu-frequenz ?

Reply:
(ah, Tiyash will probably not answer soon, so we shouldn't wait for his input.

Reply:
> IMHO UNKNOWN is definitely more accurate here, and if some estate don't have this in the spec, I think that should be a specs bug.

I think UNKNOWN was introduced for component states where we can have a bug in parsing the raw data.  That's why the the descriptions for all UNKNOWNs say "Component has state, but we understand it".  For other enums where the data doesn't come from hardware, the API doesn't have an UNKNOWN option.

But we are trying to re-purpose UNKNOWN to suggest server-client communication failure, which could happen to all enums, and that's why I think we are trying to give two meanings.

One more option is to say `UNKNOWN = -1` in all python definitions of enums that don't have unknown.

We can keep for later if you prefer, but until then, we'll have UNKNOWN and UNSPECIFIED both being repurposed for something they were not originally meant to do.

Reply:
Yeah, I think this goes beyond the microgrid API, maybe it would be a good topic for the API meeting.